lib/rubocop/cop/registry.rb

```diff
@@ -197,7 +197,7 @@ def disabled(config)
       def enabled?(cop, config)
         return true if options[:only]&.include?(cop.cop_name)
 
-        cfg = config.for_cop(cop)
+        cfg = config.for_cop(cop.cop_name)
```

Comment:
Perhaps it would be useful if a source code comment will be described about the problem avoiding the memory increase.

Reply:
Added a comment.